### PR TITLE
initial draft for a task scheduler

### PR DESCRIPTION
**Pull request by rchincha**
Signed-off-by: Ramkumar Chinchani <rchincha@cisco.com>

**What type of PR is this?**

**Which issue does this PR fix**:

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by rchincha** [comment]
For `sync` feature, add a Task and wait for callback

For 'gc/scrub', create a Task that periodically scans repos, which in turn creates a Task per-repo

**Comment by peusebiu** [comment]
will look into it :D, thanks, saw your email but didn't have time to spend on this.

File: pkg/extensions/scheduler/scheduler.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	preDelay time.Duration
+	// wait for 'postDelay' before finishing 'work'
+	postDelay time.Duration
+	// actual work, modify the signature
+	work func()
+	// once work is done, signal via this
+	cb func()
+}
+
+type Task interface {
```
Comment (peusebiu) [inline pkg/extensions/scheduler/scheduler.go:16]: Why a Task should have a Cancel method exported but not work and cb?

File: pkg/extensions/scheduler/scheduler.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
+package scheduler
+
+import "time"
+
+type task struct {
```
Comment (peusebiu) [inline pkg/extensions/scheduler/scheduler.go:5]: Why is task unexported? 
How do we use it?

**Comment by peusebiu** [comment]
> For `sync` feature, add a Task and wait for callback
> 
> For 'gc/scrub', create a Task that periodically scans repos, which in turn creates a Task per-repo

What sync callback should do?
Why do we need callbacks?

**Comment by peusebiu** [comment]
Also should we use the framework mentioned in your issue to actually schedule this, or not?

**Comment by andaaron** [comment]
No longer needed as https://github.com/project-zot/zot/pull/700 is merged.

**Pull request closed without merging**